frame/system/src/lib.rs

```diff
@@ -824,6 +830,23 @@ impl<T: Trait> Module<T> {
 		AllExtrinsicsWeight::get().unwrap_or_default()
 	}
 
+	/// Get the quota ratio of each dispatch class type. This indicates that all operational and mandatory
+	/// dispatches can use the full capacity of any resource, while user-triggered ones can consume
+	/// a portion.
+	pub fn get_dispatch_limit_ratio(class: DispatchClass) -> Perbill {
+		match class {
+			DispatchClass::Operational | DispatchClass::Mandatory
+				=> <Perbill as sp_runtime::PerThing>::one(),
+			DispatchClass::Normal => T::AvailableBlockRatio::get(),
+		}
+	}
```

Comment:
@kianenigma is this too much leaking weights into System?

Couldn't use CheckWeight struct because the trait definition was too strict.

Reply:
It is fine given that now system and weight are simply coupled together. No need to block progress now hoping to de-couple them.